### ISSUE TITLE
Stabilise outcome display

### ISSUE THREAD

**Issue by andreievg** [openmsupply/application-manager-web-app#903]
Currently outcome display for table or full construct gql query, some fields might be missing in this gql query (causing a blank display)

Proposed solution, check against schemaColums when constructing this query:

```
schemaColumns(filter: {tableName: {equalTo: "organisation"}}) {
    nodes {
      columnName
    }
  }
```

**Comment by CarlosNZ** [comment]
Yeah, so it needs to do two things:

- check if the **expected** columns (as defined in outcomes_display_table) actually exist, and ignore them if not
- see if there are any columns that exist that **aren't** defined in outcomes_display_table and try and display a default result for them (stringify whatever is in the field, perhaps?)

**Comment by CarlosNZ** [comment]
Okay, here is my suggested improvements for this, with the aims being:
- make it robust against any inconsistencies or errors
- minimise the amount of specific config required (i.e. mostly rely on defaults)
- we'd like to be able to display non-text entries in Table view sometimes (e.g. company logos)

### Column schema for "outcome_display" table

- `id`, `table_name`, `plural_table_name`, `title`, `code` -- same as current
- `detail_column_name` -- rename to "detail_title_column`
- `table_include_columns` -- array of column names to include in table display; default: NULL => include ALL
- `table_exclude_columns` -- array of column names to exclude (subtracted from "include" list)
- `detail_include_columns` / `detail_exclude_columns` -- same as above but for "details" view

Instead of "outcome_display_table" and "outcome_display_detail" tables, we just combine into one table: "outcome_display_format" (This is based on the assumption that any formatting specs for one would be the same as the other -- if there are any cases where this may differ, we can reconsider)

### Column schema for "outcome_display_format" table

Assume: most columns for each outcome tables can be easily represented in "text" form (shortText, numbers, etc.), so we ONLY need to add entries here for columns that need _special_ display (logo images, lists, etc.)

- `id`, `column_name`, `title`, `outcome_display_id` -- same as current, but with the assumption that "title" can usually be derived from column name, only needs specifying if different
- image_url -- if specified, shows an image instead of text. Used for logos/uploaded images, etc. (This is the only exception to the "Text" or "SummaryView" display -- I think it's the only one we need?)
- parameters -- as current, but only specified if we actually want to invoke a "SummaryView" element when in detail view. Default will just show Text view of field (in this case the entry probably isn't even required, since default display does this anyway) 

So basically, for the most part, everything for displaying outcome tables and details can be deduced from the "outcome_display" table, including what columns to show where. We only need to have an entry for "outcome_display_format" if there's any difference from default columnName/Text display

### Some additional logic

- If the table being queried is missing some columns defined in "include_columns" lists, we ignore them
- If the table being queried has columns NOT specified in "include_columns" lists:
  - if "include" is NULL/ALL, we attempt to include these columns in both table and details view, with default "text" formatting
  - if "include" is a finite list (i.e. not NULL/ALL), we ignore additional columns
